microcloud/service/lxd.go

```diff
@@ -420,6 +421,17 @@ func (s *LXDService) Restart(ctx context.Context, timeoutSeconds int) error {
 		return fmt.Errorf("Detected pre-existing LXD storage pools. LXD might have already been initialized")
 	}
 
+	server, _, err := c.GetServer()
+	if err != nil {
+		return fmt.Errorf("Failed to get LXD server information: %w", err)
+	}
+
+	if semver.Compare(semver.Canonical(fmt.Sprintf("v%s", server.Environment.ServerVersion)), semver.Canonical(fmt.Sprintf("v%s", lxdMinVersion))) < 0 {
```

Comment:
I think future readers would really appreciate a comment here explaining *why* we are doing this.

Reply:
Fixed.